dockerfiles/evmapp/entrypoint.sh

```diff
@@ -90,56 +100,66 @@ done
-    sleep 5
-    exit 1
-  fi
-
-  if  [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" = "true" ] && [ -z "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS:-}" ]; then
-    echo "Error: When SCNODE_CERT_SIGNING_ENABLED=true and SCNODE_REMOTE_KEY_MANAGER_ENABLED=true SCNODE_REMOTE_KEY_MANAGER_ADDRESS needs to be set."
-    sleep 5
-    exit 1
-  fi
-
-  if  [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" = "true" ] && [ -z "${SCNODE_REMOTE_KEY_MANAGER_REQUEST_TIMEOUT:-}" ]; then
-    echo "Error: When SCNODE_CERT_SIGNING_ENABLED=true and SCNODE_REMOTE_KEY_MANAGER_ENABLED=true SCNODE_REMOTE_KEY_MANAGER_REQUEST_TIMEOUT needs to be set."
-    sleep 5
-    exit 1
-  fi
-
-  if  [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" = "true" ] && [ -z "${SCNODE_REMOTE_KEY_MANAGER_PARALLEL_REQUESTS:-}" ]; then
-    echo "Error: When SCNODE_CERT_SIGNING_ENABLED=true and SCNODE_REMOTE_KEY_MANAGER_ENABLED=true SCNODE_REMOTE_KEY_MANAGER_PARALLEL_REQUESTS needs to be set."
-    sleep 5
-    exit 1
+  # Checking first for either SCNODE_CERT_SIGNERS_SECRETS not empty and SCNODE_REMOTE_KEY_MANAGER_ENABLED !=true and vice versa
+  if [ -z "${SCNODE_CERT_SIGNERS_SECRETS:-}" ]; then
+    if [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" != "true" ]; then
+      echo "Error: if SCNODE_CERT_SIGNING_ENABLED=true either SCNODE_CERT_SIGNERS_SECRETS must NOT be empty or SCNODE_REMOTE_KEY_MANAGER_ENABLED=true is required to be set."
+      sleep 5
+      exit 1
+    fi
   fi
 
-  if [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" = "true" ] && [ -n "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS:-}" ]; then
-     host="$(cut -d'/' -f 3 <<< "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}" | cut -d':' -f 1)"
-     port="$(cut -d ':' -f 3 <<< "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}")"
-     port="${port:-80}"
-     # make sure host and port are reachable
-     i=0
-     while ! nc -z "${host}" "${port}" &> /dev/null; do
+  # Checking all REMOTE_KEY_MANAGER_ENABLED parameters when SCNODE_REMOTE_KEY_MANAGER_ENABLED=true
+  if [ "${SCNODE_REMOTE_KEY_MANAGER_ENABLED:-}" = "true" ]; then
+    if [ -z "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS:-}" ]; then
+      echo "Error: When SCNODE_CERT_SIGNING_ENABLED=true and SCNODE_REMOTE_KEY_MANAGER_ENABLED=true SCNODE_REMOTE_KEY_MANAGER_ADDRESS needs to be set."
+      sleep 5
+      exit 1
+    else
+      host="$(cut -d'/' -f 3 <<< "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}" | cut -d':' -f 1)"
+      port="$(cut -d ':' -f 3 <<< "${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}")"
+      port="${port:-80}"
+      # make sure host and port are reachable
+      i=0
+      while ! nc -z "${host}" "${port}" &> /dev/null; do
        echo "Waiting for '${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}' endpoint to be ready."
        sleep 5
        i="$((i+1))"
        if [ "$i" -gt 48 ]; then
          echo "Error: '${SCNODE_REMOTE_KEY_MANAGER_ADDRESS}' endpoint is not ready after 4 minutes."
          exit 1
        fi
-     done
-   fi
+      done
+    fi
```

Comment:
This check could be the last of the three. It will take longer as it has to reach the remote host; if any of the other required env vars are not set, we can fail faster.